hoot-core/src/main/cpp/hoot/core/cmd/ConvertCmd.cpp

```diff
@@ -74,6 +72,43 @@ class ConvertCmd : public BaseCommand
 
   virtual QString getName() const { return "convert"; }
 
+  /**
+   * Return true if all the specified operations are valid streaming operations.
+   *
+   * There are some ops that require the whole map be available in RAM (e.g. remove duplicate
+   * nodes). These operations are not applicable for streaming.
+   */
+  bool areValidStreamingOps(QStringList ops)
+  {
+    // add visitor/criterion operations if any of the convert ops are visitors.
+    foreach (QString opName, ops)
+    {
+      if (!opName.trimmed().isEmpty())
+      {
+        if (Factory::getInstance().hasBase<ElementCriterion>(opName.toStdString()))
+        {
+          ElementCriterionPtr criterion(
+                Factory::getInstance().constructObject<ElementCriterion>(opName));
+          // when streaming we can't provide a reliable OsmMap.
+          if (dynamic_cast<OsmMapConsumer*>(criterion.get()) != 0)
+          {
+            return false;
+          }
+        }
+        else if (Factory::getInstance().hasBase<ElementVisitor>(opName.toStdString()))
```

Comment:
The OsmMapConsumer was a hack on the ConstElementVisitor so that it could modify the elements. The new ElementVisitor interface doesn't require that hack so I thought it best to avoid encouraging its use.